packages/e2e/suites/full/freeze.spec.ts

```diff
@@ -0,0 +1,536 @@
+  return page.getByTestId("approve-button");
+}
+
+async function approveFreeze(page: Page, wallet: Wallet) {
+  await approveButton(page).click();
+  await wallet.sign();
+}
+
+// General tests
+
+test("TC300: Navigate to freeze index button is enabled", async ({ page, switcher }) => {
+  await switcher.council(1, page);
+  const freezeButton = page.getByText("Freeze Requests");
+  await expect(freezeButton).toBeVisible();
+  await expect(freezeButton).toBeEnabled();
+});
+
+test("TC 301: Navigate to index page shows list of empty freezes", async ({ page, switcher }) => {
+  await switcher.council(1, page);
+  await goToFreezeIndex(page);
+
+  for (const kind of ["soft", "hard", "change-threshold", "unfreeze"]) {
+    await expect(page.getByTestId(`${kind}-card`).getByText("No proposals found.")).toBeVisible();
+  }
+});
+
+test("TC302: Navigate to freeze index. Shows all create buttons enabled", async ({
+  page,
+  switcher,
+}) => {
+  await switcher.council(1, page);
+  await goToFreezeIndex(page);
+
+  await page.getByTestId("soft-create-btn").isEnabled();
+  await page.getByTestId("hard-create-btn").isEnabled();
+  await page.getByTestId("change-threshold-create-btn").isEnabled();
+  await page.getByTestId("unfreeze-create-btn").isEnabled();
+});
+
+// soft freeze
+
+test("TC303, TC304: click on create soft freeze button displays correct form.", async ({
+  page,
+  switcher,
+}) => {
+  await switcher.council(1, page);
+
+  await goToFreezeIndex(page);
+
+  await page.getByTestId("soft-create-btn").click();
+
+  await expect(page.getByText("Create Soft Freeze Proposal")).toBeVisible();
+  await expect(page.getByText("Valid Until")).toBeVisible();
+  await expect(page.getByRole("button", { name: "Create" })).toBeEnabled();
+
+  await page.getByRole("button", { name: "Create" }).click();
+
+  await page.waitForURL("**/app/freeze/*");
+
+  const now = new Date();
+  const oneWeekFromNow = new Date(Date.now() + 1000 * 3600 * 24 * 7);
+  const validUntilText = await page.getByTestId("valid-until-timestamp").textContent();
+  const proposedOnText = await page.getByTestId("proposed-on-timestamp").textContent();
+  compareExtractedTextWithDate(validUntilText, oneWeekFromNow);
+  compareExtractedTextWithDate(proposedOnText, now);
+});
+
+test("TC305: Create second soft when there is an active one, fails with proper message", async ({
+  page,
+  switcher,
+}) => {
+  await switcher.council(1, page);
+  await createFreeze(page, "soft");
+
+  await goToFreezeIndex(page);
+
+  await page.getByTestId("soft-create-btn").click();
+  await page.getByRole("button", { name: "Create" }).click();
+  await expect(page.getByText("Pending proposal already exists.")).toBeVisible();
+});
+
+test("TC313: Attempt to approve soft freeze by guardian, sign button is not displayed", async ({
+  page,
+  switcher,
+}) => {
+  await switcher.guardian(1, page);
+  await createFreeze(page, "soft");
+  await assertCannotSignProposal(page, "soft");
```

Comment:
Doesn't this assert fail? As it tries to create a freeze by itself when a soft freeze already exists.